packages/story-editor/src/app/rightClickMenu/useLayerSelect.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from '@googleforcreators/react';
+import SAT from 'sat';
+import { __ } from '@googleforcreators/i18n';
+import { Icons } from '@googleforcreators/design-system';
+import { trackEvent } from '@web-stories-wp/tracking';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import useStory from '../story/useStory';
+import createPolygon from '../canvas/utils/createPolygon';
+import { useCanvas } from '../canvas';
+import { getDefinitionForType } from '../../elements';
+import { useConfig } from '../config';
+
+const ReversedIcon = styled(Icons.ChevronRightSmall)`
+  transform: rotate(180deg);
+`;
+
+function useLayerSelect({ menuItemProps, menuPosition, isMenuOpen }) {
+  const { isRTL } = useConfig();
+  const [isSubMenuOpen, setIsSubMenuOpen] = useState(false);
+  const { nodesById } = useCanvas(({ state: { nodesById } }) => ({
+    nodesById,
+  }));
```

Comment:
nit: you shouldn't need to deconstruct here since it's just `nodesById` 🤷‍♀️  
```suggestion
  const nodesById  = useCanvas(({ state: { nodesById } }) => nodesById); 
```